### PR TITLE
Use React 0.14's stateless components

### PR DESCRIPTION
**Pull request by rolyatmax**
Some spots where we could use React 0.14's stateless components.

depends on #79

### CODE REVIEW

**Comment by rolyatmax** [comment]
@AlanFoster Let me know what you think of this.

**Comment by AlanFoster** [comment]
I'm a fan of the newer notation; But we should be consistent with the rest of the code base. Particularly since this will make it harder for us to add `propType` and `defaultPropType` information in the future if we need to. Likewise newcomers to React may not fully appreciate the terseness of this PR change.

 I'll close this PR for now, but we might re-visit it in the future!

**Comment by rolyatmax** [comment]
Do you mean that to use React's stateless functions syntax, we would need to use _only_ stateless components in order to be consistent?

Or do you mean we should be consistent with how we define stateless components?:

```javascript
const Link = ({href, text}) => <a href={href}>{text}</a>;

// versus

function Link({href, text}) {
    return <a href={href}>{text}</a>;
}
```

Also, for what it's worth, you can add `propType` validation like so:

```javascript
function MyComponent({children}) {
    return <div>{children}</div>;
}

MyComponent.propTypes = {
    children: React.PropTypes.node.isRequired
};
```

File: example/index.jsx [hunk 1]
```diff
@@ -87,109 +87,103 @@ const DatePickerSingle = React.createClass({
 var mainCodeSnippet = fs.readFileSync(__dirname + '/code-snippets/main.jsx', 'utf8');
 
 
-const Index = React.createClass({
-  getDefaultProps() {
-    return {};
-  },
```
Comment (rolyatmax) [inline example/index.jsx]: If we don't want to move ahead with this PR, should we pull this change out?

**Pull request closed without merging**